### PR TITLE
[SYCL][libdevice] Add -fsycl-device-lib-online-link option.

### PR DESCRIPTION
**Pull request by jinge90**
When this option is used, we use online link for SYCL device libraries,
otherwise, we use offline link, all wrapper and fallback device libraries
are linked with user's device image in compilation time.

Signed-off-by: jinge90 <ge.jin@intel.com>

### CODE REVIEW

**Comment by pvchupin** [comment]
@jinge90, can you please add update of https://github.com/intel/llvm/blob/sycl/sycl/doc/UsersManual.md into this patch?

File: clang/include/clang/Driver/Options.td [hunk 1]
```diff
@@ -4854,6 +4854,12 @@ def fno_sycl_device_lib_EQ : CommaJoined<["-"], "fno-sycl-device-lib=">, Group<s
   Values<"libc, libm-fp32, libm-fp64, all">, HelpText<"Control exclusion of "
   "device libraries from device binary linkage. Valid arguments "
   "are libc, libm-fp32, libm-fp64, all">;
+def fsycl_device_lib_online_link : Flag<["-"], "fsycl-device-lib-online-link">,
+  Group<sycl_Group>, Flags<[NoArgumentUnused, CoreOption]>, HelpText<"Enables "
+  "sycl device library online link">;
```
Comment (pvchupin) [inline clang/include/clang/Driver/Options.td:4859]:
```suggestion
  "sycl device library online link (experimental)">;
```

File: clang/include/clang/Driver/Options.td [hunk 2]
```diff
@@ -4854,6 +4854,12 @@ def fno_sycl_device_lib_EQ : CommaJoined<["-"], "fno-sycl-device-lib=">, Group<s
...
+def fno_sycl_device_lib_online_link : Flag<["-"], "fno-sycl-device-lib-online-link">,
+  Group<sycl_Group>, Flags<[NoArgumentUnused, CoreOption]>, HelpText<"Disables "
+  "sycl device library online link">;
```
Comment (pvchupin) [inline clang/include/clang/Driver/Options.td:4862]:
```suggestion
  "sycl device library online link (experimental)">;
```

File: clang/include/clang/Driver/Options.td (hunk 1 above)
Comment (jinge90) [reply clang/include/clang/Driver/Options.td]: Done.

**Comment by jinge90** [comment]
> @jinge90, can you please add update of https://github.com/intel/llvm/blob/sycl/sycl/doc/UsersManual.md into this patch?

Doc added.

File: sycl/doc/UsersManual.md [hunk 3]
```diff
@@ -191,6 +191,14 @@ and not recommended to use in production environment.
     libm-fp32, libm-fp64, libc, all. Use of 'all' will enable/disable all of
     the device libraries.
 
+**`-f[no-]sycl-device-lib-online-link`** [EXPERIMENTAL]
+
+    Enables/disables online link mechanism for SYCL device library in JIT
+    compilation. If online link is enabled, all required device libraries will
+    be linked with user's device image by SYCL runtime during execution time,
+    otherwise the link will happen in build time, online link is disabled by
+    default currently and this option is ignored in AOT compilation.
```
Comment (pvchupin) [inline sycl/doc/UsersManual.md:200]:
```suggestion
    default currently. This option is ignored in AOT compilation.
```
nit

**Review by pvchupin — DISMISSED** [review]

**Comment by jinge90** [comment]
Hi, @mdtoguchi 
Could you help review this patch?

Thanks very much.

**Review by mdtoguchi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**